[App Extensibility] Sync with latest `develop` branch (@W-18121817@)

vmarta opened the pull request:
Merging in the latest changes from `develop` branch. Just a few commits:
- 4b503f7ce3775ef1c8a153a6399090395bc400d4
- d86dc7bea3fe03a0f475ecf1a0ab878374f07026
- 76fc5b0a4e8c2c0426eb54d0bbba8c95fad8cfe4
- #2375

alexvuong committed:
W-17908517 - Fix basket loading after logging out (#2323)

* fix basket loading after logging out

vcua-mobify committed:
[Snyk] Dependency updates (#2338)

* Update nanoid dependency

* Override path-to-regexp version

* Update CHANGELOG.md

* Apply snyk version bumps

* Update changelogs

* Update CHANGELOG.md

* Update changelogs

vmarta committed:
Sync with latest develop branch

vmarta committed:
Update package lock files

cc-prodsec commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/a54ead7e-a6c9-4f5a-8612-e1dabf57f0b8?refs=5fd80e2e9267928c75e2051c389d121439a428b2&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/a54ead7e-a6c9-4f5a-8612-e1dabf57f0b8/license?refs=5fd80e2e9267928c75e2051c389d121439a428b2&source=prChecksComment)

alexvuong committed:
@W-13747172 Fix a11y (#2375)

* fix a11y

vcua-mobify committed:
@W-17830285@ Add custom parameters to SLAS authorize calls (#2358)

* Allow custom parameters to SLAS authorize calls

* Update CHANGELOG.md

* Move extractCustomParameters to utils

* Use options and body instead of parameters for consistency

* Ensure only custom parameters are set from register to login

* Don't send custom params to shopperCustomer register

* Allow custom params to be set via login.mutateAsync

* Update tests using ShopperCustomers

* Apply feedback

* Add comments

* Lint

vmarta committed:
Pin to previous version of cc-datacloud-typescript

Their latest 1.1.0 release that's out today was causing an error with building our pwa-kit project.

vmarta committed:
Revert "Pin to previous version of cc-datacloud-typescript"

This reverts commit e558f520a94d6a4788c0ab97a2bd6c97da01e251.

vcua-mobify reviewed (approved):
+1 as the builds are now passing.

Also, there's a new PR from last week that we might want to add either as part of this or in a follow up: https://github.com/SalesforceCommerceCloud/pwa-kit/commit/76fc5b0a4e8c2c0426eb54d0bbba8c95fad8cfe4

shethj committed:
Committing temp version bump

shethj committed:
Revert "Committing temp version bump"

This reverts commit 398cfce2ecd63197363f4694757dee9fc924273d.

vmarta committed:
Merge branch 'develop' into vm/sync-extensibility-with-latest-develop

vmarta committed:
Update package lock files

vmarta committed:
Try replacing `sh.cat` with native file reading

See if such replacement works well on a Windows machine.

vmarta committed:
Debug: see what gets generated

vmarta committed:
Debugging

vmarta committed:
Refactor file copying logic

vmarta committed:
Clean up

vmarta committed:
Revert "Try replacing `sh.cat` with native file reading"

This reverts commit dc26b6e39c95d3e0dd1743e04fd50779f968bd08.

vmarta committed:
Remove debug code

vmarta commented on packages/pwa-kit-create-app/scripts/create-mobify-app.js:
```diff
@@ -878,10 +878,7 @@ const processAppExtensions = (
-            sh.cp('-rf', p.join(appExtensionTmpPath, '.*'), appExtensionDestDir)
-            // Copy regular files
-            sh.cp('-rf', p.join(appExtensionTmpPath, '*'), appExtensionDestDir)
+            copyAllFiles(appExtensionTmpPath, appExtensionDestDir)
```
The changes in this file are new.. they're not from the `develop` branch. They fix the failing project generation on windows machine.

Otherwise, you would see an error like this:

![Arc 2025-04-30 at 22 39 09](https://github.com/user-attachments/assets/d904feb9-62e9-4f5f-8d52-dadf27738086)

vcua-mobify replied:
Should we add this change and the ones you've noted below back to the develop branch afterward?

vmarta replied:
@vcua-mobify Good question. Fortunately we don't need to bring those changes back to develop. They're only needed for the extensibility branch as the issue is newly introduced there.

vmarta commented on packages/pwa-kit-create-app/scripts/create-mobify-app.js:
```diff
@@ -989,7 +1000,7 @@ const runGenerator = async (
 
     // Copy the base template either from the package or npm.
-    sh.cp('-rf', p.join(packagePath, '{*,.*}'), outputDir)
+    copyAllFiles(packagePath, outputDir)
```
A similar problem here on Windows machine. The error looks like this:

![Arc 2025-04-30 at 22 17 40](https://github.com/user-attachments/assets/dbd97a45-1b40-4a1d-8e1c-5d178ba3ab28)

vmarta commented on packages/pwa-kit-create-app/scripts/create-mobify-app.js:
```diff
@@ -939,7 +950,7 @@ const fetchAvailableAppExtensions = () => {
  * @param {*} answers
  * @param {*} param2
  */
-const runGenerator = async (
```
No need to have `async` here actually. 

In fact, it was causing a weird issue where an error was not surfaced. An error inside the `runGenerator` function was strangely not caught at the end (see line 1314). 

So it looked as if everything was fine, and a successful message was printed out. This is probably why the failing project generation on Windows machine was not noticed until now.

vcua-mobify commented:
Just noting that it looks like this PR also contains https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2375

Could you update the pwa-kit-create-app changelog @vmarta to note the new changes?
Also, there's no changelog yet for extension-chakra-storefront

vmarta committed:
Add changelog for the chakra-storefront

vmarta committed:
Update CHANGELOG.md
